### ISSUE TITLE
Quasar v2 (Vue 3)

### ISSUE THREAD

**Issue by rstoenescu** [quasarframework/quasar#7836]
**Quasar v2 is out!** Read our [announcement](https://github.com/quasarframework/quasar/discussions/9734).

## Is v2 stable?

**YES**

## The plan

Our main focus at the moment is shipping Quasar v2 with Vue v3.
The plan is to **not introduce any breaking changes**, unless forced by the vue 3 / vue-router 4 architectures.

Affected Quasar packages: "quasar", "@quasar/app".

## Rough schedule

* First beta: "quasar" v2.0.0-beta.1 & "@quasar/app" v3.0.0-beta.1 - **Jan 2021**
* Stable releases: "quasar" v2.0.0 & "@quasar/app" v3.0.0 - **Q2 2021**

## The code

The branch currently holding Quasar v2 is called [dev](https://github.com/quasarframework/quasar/tree/dev) (was previously `vue3-work`).

## Donations

If you want to speed up the development of Quasar please consider donating to the project. With proper funding, it allows more of the team to work on the project in a much more dedicated manner.

[Donations - https://donate.quasar.dev](https://donate.quasar.dev)

If you're in a company and using Quasar for commercial projects, explain to your management the importance of monthly donations (eg. $200+) for open source projects: you're the one using it every day and this makes you the best suited person to convince them. Be creative! :slight_smile:

Quasar is saving your company tens of thousands of development hours (quite literally), which in turn is a pretty high money savings. Consider giving back a part of those savings to refuel the project itself :wink:

## Update: June 21st 2021

Quasar v2 stable (with Vue.js 3) has been released! Time to party!
https://github.com/quasarframework/quasar/discussions/9734

## Update: June 19th 2021

Next release is v2.0.0 stable.
Branch `vue3-work` has been merged into `dev`. The Qv1 work will now be done exclusively on the new `v1` branch.

## Update: April 27th 2021
We have passed the final major milestone: Quasar CLI now ships with Webpack 5, which was badly needed to support Node 13+. Faster build speeds and lower memory usage.
We're just a few small steps away from getting Quasar v2 out of beta!

## Update: April 4th 2021
The last Quasar CLI mode left to be enabled, which is SSR, is now also available.
Here is the link to the [upgrade guide](https://next.quasar.dev/quasar-cli/developing-ssr/ssr-upgrade-guide) on SSR.
We are targeting for a 2 weeks testing period for SSR mode then we will release the official stable version of Quasar v2.

## Update: February 16th 2021

We're close to the stable release. The only thing remaining is the SSR support. Quasar is SSR ready, however we're only going to enable SSR support as soon as these Vue3 SSR bugs (critical for us and your experience) are solved:
* ~https://github.com/vuejs/vue-next/issues/3244~ redirected to https://github.com/vuejs/vue-next/issues/2863
* https://github.com/vuejs/vue-next/issues/2386
* https://github.com/vuejs/vue-next/issues/3245
* https://github.com/vuejs/vue-next/issues/3243
* https://github.com/vuejs/vue-next/issues/3255

## Update: February 10th 2021

SSR mode is ready (UI and CLI-wise) to be released. Unfortunately, there are quite a few Vue SSR bugs that need to be fixed before we enable the build mode for Quasar v2. We are in the process of reporting them and we'll start collaborating with the Vue team to help fix them.

## Update: January 11th 2021

Hi All,

There are still some things being worked on for Quasar v2, but mainly we're now writing the migration guide. Our target for release of the public beta is within January 2021. The new version is based on Composition API. But it's not just a port to Vue 3. _There are lots of significant performance enhancements in Quasar's algorithms too!_ **You'll love it!**

Unfortunately, the first beta release won't include SSR support, due to missing some upstream features in Vue 3, which we'll contribute to via PRs. We'll release SSR support during the beta stage lifecycle though.

We're also trying to integrate Webpack 5 support into the `@quasar/app` v3 release, but if we are unable to get this done (ie: external dependencies and such) then we'll shoot for `@quasar/app` v4 release.

Stable release will be anywhere within Q1 2021, but as is the case with OSS, it will be ready, when it's ready. And of course, it will be ready when we know the quality is high enough to meet both your expectations and ours.

In the meantime, the wonderful framework you've come to love is still here and you don't have to wait for Quasar v2 to start a new project with Vue 3's major new feature (i.e. using the composition API). Of course, if you choose to hold off on Vue 3 features, we'll make the upgrade as easy as possible for you. That's a promise. :slight_smile:

The Quasar Team

**Comment by ontwikkelfabriek** [comment]
@rstoenescu 
Will the codebase of Quasar v2 with Vue 3 be rewritten in TypeScript just like Vue3 is?

**Comment by IlCallo** [comment]
Quasar v2 will lay the fundations needed for a Composition API + TS + tests incremental migration, which will take place during Quasar v2 lifecycle.

Quasar v3 will probably be released when this incremental migration is completed.

**Comment by chintan-mishra** [comment]
Those who downvoted. Or are planning to do so. READ THIS.

Providing a stable API that maintains as much compatibility as possible requires time, effort, and capital. This is not to say that things won't break when moving from v1->v2. But the collateral effect of migrating to a complete TS based Quasar will be too large on the wider community. Thus, as an engineer I am following the exact same steps mentioned here. I am migrating some code of my very large Quasar v1 with Vue2 to TS and by the time v3 arrives we will have the whole community using TS.

Finally, if you are creating a new project and want to use TS then just add TS support when creating the application using the wizard.

**Comment by rstoenescu** [comment]
Maybe an additional important information. For app developers it won't matter whether Quasar is written in TS or not. Sure, the autocomplete will work a bit better, but you as a developer are NOT required to also use TS in your app.

**Comment by msinkgraven** [comment]
Are there any plans or ideas to support vite in quasar v2?

**Comment by zhengycsh** [comment]
desire

**Comment by leannc** [comment]
Is the beta version available now ? Waiting for this~

**Comment by rstoenescu** [comment]
Hi everyone,

Sorry but we'll have to delay the beta version due to vue-loader which has some SSR features not implemented yet. We cannot release a beta version until SSR is fully solved or else we might introduce breaking changes during this beta stage -- and we want to avoid this at all costs.

The new schedule is:

* First beta: "quasar" v2.0.0-beta.1 & "@quasar/app" v3.0.0-beta.1 - **2020/12/15**
* Stable releases: "quasar" v2.0.0 & "@quasar/app" v3.0.0 - **2021/01/30**

**Comment by jeroenqui** [comment]
Hello @rstoenescu 
Thanks for your hard work, I've been following the vue3 progress closely and with interest. Nice job so far! Of course we will wait patiently for the beta release. Can you give more details about which SSR features are not implement in the vue-loader yet?

**Comment by bzd2000** [comment]
Is there an Alpha release?

**Comment by rdev06** [comment]
Can I use branch [vue3-work](https://github.com/quasarframework/quasar/tree/vue3-work) for learning Quasar with vue 3 purpose?

**Comment by bzd2000** [comment]
Tried to build the vue3-work branch, but don't see how to build to app package.
I followed to instructions in the contribution guideline, but it seems only for the UI package.

Are there instruction on how to build the App package?

**Comment by BenHayat** [comment]
@rstoenescu
I had recently made a suggestion on next Quasar version # in the forum. I'd like to echo that here as well.

Most third party products try to follow the main product version # to make it more clear to the customers. For example, now the the new .Net 5 is out, most of the third party libraries, they use # 5 as well. Like Entity Framework 5 or ASP.Net 5 and etc.
In Vue, Vuetify and PrimeVue are doing the same following V3.
My suggestion is to skip Quasar V2 numbering and go straight to V3. This way, new prospects won't get the wrong impression that Quasar V2 is for Vue 2.
Hope this helps.

**Comment by smolinari** [comment]
@BenHayat - Unfortunately, that isn't going to work. Quasar/UI v2 is planned to be purely a port to Vue 3. Quasar v3 is planned to be a further and important evolution of Quasar, also built on Vue 3. See the dilemma? 😊 

Scott

**Comment by BenHayat** [comment]
> 
> 
> @BenHayat - Unfortunately, that isn't going to work. Quasar/UI **v2 is planned to be purely a port to Vue 3**. **Quasar v3** is planned to be a further and important evolution of Quasar, also **built on Vue 3**. See the dilemma? 😊
> 
> Scott

Now it all makes sense. Quasar V2 and Quasar V3.
I didn't know this information.

Thank you Scott!!!

**Comment by layanto** [comment]
Will Quasar V2 and V3 consider supporting Pinia as alternative to Vuex? I think the API for Pinia is similar to the proposed API for Vuex 5 and has better typescript support.

**Comment by smolinari** [comment]
@layanto - Vuex isn't directly a part of Quasar. It's only suggested during project creation via the CLI. In the end, you can use whatever state management package you'd like, as long as it works with Vue.

Scott

**Comment by layanto** [comment]
What about accessing store from boot file if using Pinia as state management instead of Vuex?

**Comment by layanto** [comment]
What about accessing store from boot file if using Pinia as state management instead of Vuex?

**Comment by smolinari** [comment]
I believe Quasar will inject in anything found under `src/store`. @layanto Don't quote me on that though.  😁 

Scott

**Comment by rstoenescu** [comment]
Teaser: https://codepen.io/rstoenescu/pen/dypbRPL?editors=111

**Comment by JulesNeck** [comment]
This looks really promising. The new quasar version will be my favourite christmas gift. Thanks Razvan for your magnificent work!

**Comment by BenHayat** [comment]
What is the significance of this teaser, beside these fact that Quasar is running in V3? Am I missing something from looking at those cards?

**Comment by danielfaust** [comment]
@BenHayat That finally the migration of that old AngularJS-webapp to Vue3+Quasar is within reach?

**Comment by smolinari** [comment]
@BenHayat - It's just a sign.. a teaser.. that Quasar v2 is getting closer to being a reality. 

Scott

**Comment by BenHayat** [comment]
> 
> 
> @BenHayat - It's just a sign.. a teaser.. that Quasar v2 is getting closer to being a reality.
> 
> Scott

There is absolutely no doubts in anyone's mind that the team will deliver even better than others.
I thought I was missing something in particular, and that's why asked.

Thanks Scott

**Comment by JulesNeck** [comment]
**@BenHayat**
The new composition api of Vue 3 makes it possible to seperate code by logical concerns instead of spreading it over options and mixins. So it's just a different way to write code. The new quasar version will make it easier to write larger and more complex applications. But I guess you already know this. For me as a developer working on a quite complex project with quasar this will impact a lot of my code. The teaser isn't spectacular, granted. But it shows that Razvan is busy working on it. And this are very good news.

**Comment by BenHayat** [comment]
> 
> 
> **@BenHayat**
> The teaser isn't spectacular, granted. But **it shows that Razvan is busy working on it. And this are very good news.**

Thanks! I got the same impression and I'm very happy. Honestly, there isn't anything out there that can get close to Quasar. Trust me I've tried them... :-)

**Comment by bzd2000** [comment]
Indeed, it is the most complete and best maintained.

**Comment by danielfaust** [comment]
> there isn't anything out there that can get close to Quasar

I've tried them as well. For me the one thing which immediately hooked me on Quasar was the fact that the popup dialog would not scroll the page if dragged, it really kept a lock in the pages scrolling position, and if anything would scroll, it would be the content of the dialog, if designed to scroll, but nothing else. It's a very minor thing, but it shows the level of detail in correctness of this framework.

**Comment by JulesNeck** [comment]
@BenHayat
Of course I trust you! I also tried a lot of different tools and frameworks over the years. I started with Visual Basic .Net, after that C#. I even used Angular and react a bit. Before Typescript came up I worked on a huge project with Coffeescript and the Qt framework (which is basically C++). I also experimented with ionic. And after that I discovered quasar. It's like a developers dream! It's so elegant and clean. Even the code looks nice... ;-)

**Comment by BenHayat** [comment]
@rstoenescu ;
Hi Razvan;
Could you please clarify something for me?
In Quasar V2, does this mean that we can create a Vue 3 project, install Quasar V2 and write Vue 3 code, i.e. composition API, even though the Quasar components haven't been converted/re-written for Vue Composition API?

At this point, it's not important to me, if the Quasar components are NOT re-written for Vue, as long as I can build Vue compatible project and use Vue 3 features and Quasar components.

Thank you in advance!

**Comment by rstoenescu** [comment]
@BenHayat Correction. More than half of Quasar components have been rewritten using Composition API. Eventually they will all be (it's my current focus these days). But regardless, you'll be able to use Options API or Composition API or any other Vue 3 feature in your projects.

**Comment by smolinari** [comment]
@BenHayat 
> does this mean that we can create a Vue 3 project, install Quasar V2 and write Vue 3 code,

The suggested and better route will be, create a Qv2 project (via the Quasar CLI) and develop your app with Vue 2 and/ or Vue 3 concepts (i.e. options or composition APIs), but have your app 100% driven by Vue 3 at the core. 😁 

Scott

**Comment by BenHayat** [comment]
@rstoenescu 
> 
> 
> @BenHayat **_Correction. More than half of Quasar components have been rewritten using Composition API_**. Eventually they will all be (it's my current focus these days). But regardless, you'll be able to use Options API or Composition API or any other Vue 3 feature in your projects.

Hi Razavan;
Previously, Scott had mentioned:
> Quasar/UI v2 is planned to be purely a port to Vue 3
And when I asked for clarification from you, I was mainly trying to understand what Quasar 2 is and how we can use it with Vue 3.
But you said "Correction. More than half of Quasar components have been re-written using Composition API"

So, now I'm even more confused.
Can we have a clear statement, on the path how we should peruse with NEW project?

I'm getting conflicting answer between Razavan's and Scott's answers.
Thanks!

**Comment by rstoenescu** [comment]
What Scott and I said is not conflicting at all. Quasar v2 will be based on Vue 3.
Vue 3 offers both Options API and Composition API (although Composition API is preferred), so you can write your components in the form that you like. Options API is not something specific to Vue 2 only.
It's irrelevant whether Quasar UI is written with Composition API or not as it does not affect your devland code.

You will be able to upgrade your current project to use the newer "quasar" and "@quasar/app" versions or you can create a new Quasar CLI project. The stride with Quasar v2 is to have a smooth transition when developers upgrade to it. All .vue files will mostly not need any update since vue-loader will compile SFC directly to Vue 3. There are a few (but NOT many) breaking changes in Quasar UI as well (due to the under the covers Vue3 architecture change), but upgrading should be very easy.

**Comment by BenHayat** [comment]
@rstoenescu 

>What Scott and I said is not conflicting at all. Quasar v2 will be based on Vue 3.

But, those two answers, were related to porting existing Quasar Vue 2 apps to Quasar Vue 3 app. That's not what I've asking.

> It's irrelevant whether Quasar UI is written with Composition API or not as it does not affect your devland code.

That was my point in my original question, which I said, "It's not important to me if Quasar's components are written in Option API or Composition API". So we both agree on that.

My main question was, which you kind of answered, for a NEW project, Can we start with Vue 3 and composition API for OUR components and NOT write the code with Option API (Vue 2 style) and have to go back again and change all the Option API codes to Composition API? This is what I'm trying to avoid.
My question isn't about porting Quasar 1.x & Vue 2 code to Vue 3 code. It's about starting a NEW project using Composition API for our components.

I hope my question is more clear now.

**Comment by rstoenescu** [comment]
@BenHayat to summarize: yes, you can start with composition api in a new project

**Comment by smolinari** [comment]
@BenHayat - Not trying to sound like a smartass here, but

> Quasar/UI v2 is planned to be purely a port to Vue 3

If Quasar is porting to Vue 3, it would be majorly silly for us to have the Quasar v2 users have new projects only working in a Vue 2 fashion (i.e. only offer the options API). So, to me, it was common sense that a port to Vue 3 for Quasar means Quasar v2 users would get all the bells and whistles of Vue 3 i.e. they can use the composition API. Sorry, if that assumption caused the confusion. 

Scott

**Comment by BenHayat** [comment]
> 
> 
> @BenHayat - Not trying to sound like a smartass here, but
> 
> > Quasar/UI v2 is planned to be purely a port to Vue 3
> 
> If Quasar is porting to Vue 3, it would be majorly silly for us to have the Quasar v2 users have new projects only working in a Vue 2 fashion (i.e. only offer the options API). So, to me, it was common sense that a port to Vue 3 for Quasar means Quasar v2 users would get all the bells and whistles of Vue 3 i.e. they can use the composition API. Sorry, if that assumption caused the confusion.
> 
> Scott

In the past, I've been burnt by making an assumption based on "Common sense" or "my interpretation". Now I have to be very careful and clear, as my decision affects others and I'm responsible for that.
So I tend to be very black & white not to fall into my own wrong assumption's trap.

**Comment by okssq** [comment]
Today is December 15, has the plan been postponed?

**Comment by pdanpdan** [comment]
@okssq Where are you from? :)

**Comment by okssq** [comment]
@pdanpdan 
I would like to ask when will the beta version be released? 0.0

**Comment by rstoenescu** [comment]
Sorry, but postponed for January. We're not yet at the level that we'd like.

**Comment by okssq** [comment]
Hard work, everyone.thank you everyone! Is there a certain time, otherwise I have to find something else because of work

**Comment by IlCallo** [comment]
Hey @okssq, consider that the deadlines written here are for the *beta* version of Quasar 2.
If you have to start a new production project, you'd want to wait for the stable release and use Quasar 1 in the meantime: we'll provide an upgrade guide when Quasar 2 is ready :)

**Comment by okssq** [comment]
我看到了，我本来是想先用beta version of Quasar 2开始写这个项目，等项目完成quasar的正式版估计也差不多完成了。但是现在quasar 1编程vue 3好像不是很友好！

**Comment by okssq** [comment]
I saw it. I originally wanted to use the beta version of Quasar 2 to start writing this project. When the project is completed, the official version of Quasar is estimated to be almost complete. But now quasar 1 programming vue 3 does not seem to be very friendly!

**Comment by robertzml** [comment]
I'm waiting for the beta version of Quasar 2 to start a new project!  Alpha release is also acceptable.

**Comment by danielfaust** [comment]
I'm also waiting for the beta version in order to start a migration from Vue2 to Vue3, so I guess I'm going to start either in January or see how I can get started with the demo code which @rstoenescu provided in order to start earlier in order to get my feet wet.

I have no need to rush into production, as the development/migration will take its time anyway. Also, this gives me the opportunity to provide feedback while the beta is running.

The greatest thing is about this thread is that it is now clear that Quasar will not only be available for Vue3, but that this will happen relatively soon.

Thanks to all who create this excellent framework.

**Comment by bonatoc** [comment]
Not meaning to be a pain in the ass, but given how excellent framework this is, I think a "hello" and "please" would be appropriate from some folks who come and say "is it ready yet? Is it ready yet?", as if "Open Source" meant not considering all the hard work and dedication that is at hand.

I'm going into beta real soon, and if all goes well, I plan to [chime in](https://github.com/sponsors/rstoenescu).

Saying thank you is all fine and dandy, but let us not forget that without dineros, we ain't gonna go far.
And this framework deserves the same success Nuxt has.

Maybe just increment the font size of the doc would help.
As it is, it gives the impression of being too much "developer" oriented, if this makes any sense. 
I get a slight a "nerd" feeling from it.

Also, subcategories for UI components would be great.
The UI components are fantastic, but only "Forms" gets the subcategory treatment,
making browsing them a bit tedious.

Aside these minors complaints, I've been suprised at how fast it is, at how easy a default PWA was built, and how well Pug handles the components (even though acquiring Pug reflexes is kind of a steep curve).

Congratulations for the excellent work.
I was impressed at the docs when I first discovered Quasar, I am even more impressed that none of them stated bullshit. This beast delivers.

**Comment by zcaudate** [comment]
Hahaha. Is it ready yet? Please?

Eagerly anticipating the next release.

**Comment by BenHayat** [comment]
> 
> 
> Hahaha. Is it ready yet? Please?
> 
> Eagerly anticipating the next release.

You should read Razvan comment from four days ago.

**Comment by metalsadman** [comment]
if you want to adapt composition api in your current quasar v1 project, you can use the https://github.com/vuejs/composition-api plugin.

**Comment by IUnknown68** [comment]
Just a small note to all of you quasar guys: You did a great job so far, I'm always astonished how complete, yet flexible and open quasar is. So if you say, Q2 needs more time, then this is how it is, and we consumers of quasar should be (and I am) happy that you don't let quality slide just to keep some deadlines. So: Take all the time you need, who doesn't like to wait is most likely very welcome to help.
Way to go!

**Comment by Daniel-Uzcategui** [comment]
Hello from Venezuela to the Quasar team and community, I wanted to chime in to say thank you for all of your hard work, I have been using and learning Quasar since March, I was a F7 and vanilla JS developer, now with Quasar and the Help of Udemy paid tutorials I have come a long way into the reactivity and Framework equipped projects... Right now I don't have much money to spare, I have not been employed since the beggining of the pandemic, but my time has not gone to waste, as soon as I start making money again with the app I developed through the pandemic, I will surely be able to donate a monthly percentage. Cheers Happy Holidays.

**Comment by Mausino** [comment]
Hello from Slovakia, and also i want say thank you for all of your hard work

One small question.. Is the schedule time/plan also for official extensions that will compatible with quasar v2 (Vue 3) ???
How will situation with unofficial extensions which are compatible with quasar v1 but doesn't with v2 ???

**Comment by mazraara** [comment]
When is the first beta version planned to be released in January 2020 ?

**Comment by ekoeryanto** [comment]
please move on brotha, @mazraara!
it is 2021 now 🤣

**Comment by ekoeryanto** [comment]
Hi Quasar team. I hope the next version also support vite.

Thanks.

**Comment by kotoOo** [comment]
Any updates of expectations? 
Thank you.

**Comment by hawkeye64** [comment]
@kotoOo Hopefully there will be an announcement sometime next week.

**Comment by psi-4ward** [comment]
Ive hacked the vue cli plugin a bit and use q next with v3 for some weeks now. Works great so far

**Comment by rstoenescu** [comment]
Hi All,

There are still some things being worked on for Quasar v2, but mainly we're now writing the migration guide. Our target for release of the public beta is within January 2021. The new version is based on Composition API. But it's not just a port to Vue 3. _There are lots of significant performance enhancements in Quasar's algorithms too!_ **You'll love it!**

Unfortunately, the first beta release won't include SSR support, due to missing some upstream features in Vue 3, which we'll contribute to via PRs. We'll release SSR support during the beta stage lifecycle though.

We're also trying to integrate Webpack 5 support into the `@quasar/app` v3 release, but if we are unable to get this done (ie: external dependencies and such) then we'll shoot for `@quasar/app` v4 release.

Stable release will be anywhere within Q1 2021, but as is the case with OSS, it will be ready, when it's ready. And of course, it will be ready when we know the quality is high enough to meet both your expectations and ours.

In the meantime, the wonderful framework you've come to love is still here and you don't have to wait for Quasar v2 to start a new project with Vue 3's major new feature (i.e. using the composition API). Of course, if you choose to hold off on Vue 3 features, we'll make the upgrade as easy as possible for you. That's a promise. :slight_smile:

The Quasar Team

---

If you want to speed up the development of Quasar please consider donating to the project. With proper funding, it allows more of the team to work on the project in a much more dedicated manner.

[Donations - https://donate.quasar.dev](https://donate.quasar.dev)

If you're in a company and using Quasar for commercial projects, explain to your management the importance of monthly donations (eg. $200+) for open source projects: you're the one using it every day and this makes you the best suited person to convince them. Be creative! :slight_smile:

Quasar is saving your company tens of thousands of development hours (quite literally), which in turn is a pretty high money savings. Consider giving back a part of those savings to refuel the project itself :wink:

**Comment by CyrilBLA** [comment]
Just a small comment to say I love your work and wish you the best for 2021 !

Quasar with vue 3 is just around the corner and I know you're working hard on it ! Go go go !

**Comment by BenHayat** [comment]
Friends, let's all support Quasar with monthly donations as if it was a subscription based product or an annual fee product. The team is putting their valuable resources in building this product and we need to support the team. 

I used to use a product that the beginning of each renewal year, we had to pay $2,000 for the tool, plus runtime renewals. That was the only way, for that company to sustain the expenses by enforcing those fees and everyone paid.
Now, Quasar team is sustaining these expenses out of their pocket, and sooner or later it will dry out, and we will end up with a stale product. **Let's protect Quasar which is our bread & butter.**

..Ben

**Comment by pjmuller** [comment]
Hi and also from Belgium a big "respect" for the current Quasar ecosystem.

I scrolled through this topic and did not find a clear answer yet, @rstoenescu / @hawkeye64  how much will the quasar html markup change from v1 to v2?

or in other words, to already start our project now, does it make sense to prototype the UI parts of our app using current v1 html snippets and then be able to copy paste it into a Quasar v2 project?

Greetings,
PJ

**Comment by hawkeye64** [comment]
@pjmuller Don't be afraid to start new projects. There will be a comprehensive upgrade guide that will tell you all the changes that need to be made. We hope to have an announcement out soon.

**Comment by danielfaust** [comment]
@hawkeye64 I think the bigger issue with starting a new project right now is that you probably don't want to start one by using Vue 2, when you know that you will then need to rewrite it for Vue 3. It would be better to start with Vue 3 and using the beta or whatever already exist from then next version of Quasar. So waiting for the beta will likely be the best thing to do.

**Comment by IlCallo** [comment]
@danielfaust Many Vue3 breaking changes are minor, unless you need a very specific new feature.
If you mean using Composition API instead of Options API, there is a [Vue2 package](https://github.com/vuejs/composition-api) which allow you to use the former, while the latter will be supported with Vue3 too

**Comment by BenHayat** [comment]
> 
> 
> @danielfaust Many Vue3 breaking changes are minor, unless you need a very specific new feature.
> If you mean using Composition API instead of Options API, there is a [Vue2 package](https://github.com/vuejs/composition-api) which allow you to use the former, while the latter will be supported with Vue3 too

Thanks for the info.

**Comment by vitorhugosg** [comment]
Very nice!


Very good, guys if you don’t ask for much, if you can put the prefetch of the SSR with access to this as in nuxt, it would help the framework become even more perfect <3

**Comment by hawkeye64** [comment]
@vitorhugosg The SSR won't be available in initial beta release. Because Vue3 took out what's really needed for SSR (except for some hooks), it's now up to vendors to supply it.

**Comment by bompus** [comment]
@hawkeye64 @rstoenescu You should check out vitejs/vite - they've already added SSR support, pending documentation, and vite is lightning fast for both hmr dev and build compared to webpack. I sure hope Quasar will consider offering both vite and webpack5 support when the time comes.

**Comment by IlCallo** [comment]
While we'll definitely look at vite at some point, right now we don't have bandwidth to squeeze it into quasar 2 release :)

On SSR side, surely many concepts can be reused, but it's definitely not a copy/paste of code to make it work on Quasar too. We'll try to add it as soon as possible :)

**Comment by KevinExtremo** [comment]
This is absolutely great news and I've been anxiously following this issue for a while.

Now before I go asking my question I just want to make sure it is absolutely clear I do not want to create any form of pressure. Please have it done when you feel it is ready so it can have the quality we're all accustomed to.

But since we're on the last day of January and I am about to start a project, I am very tempted to wait for the beta to try it immediately. But I get the feeling that the beta wont make January.

Do you have any new estimates yet which schedule you believe you will make since the post mentioned you're on schedule? Or can the truly daring of us even try the vue-work branch?

Otherwise would you recommend starting a quasar project with vue2 and the composition API you posted at the top to minimize migrations?

Thanks so much and I absolutely love the framework :+1:

**Comment by BenHayat** [comment]
@KevinExtremo ;

I'll share my two cents with you.
I've been going through this whole thing about should I start with Vue 2 or 3 so many times, and after literally creating a comparison list and reflecting on years of being on cutting/bleeding edge, this is the first time I feel I'll wait for Vue 3.1 to come out to make sure all supporting libraries have caught up to Vue 3. It's not just Quasar, but a Vue project uses so many internal libraries.

Right now I'm building an app with Vue 2 and Quasar 1.x and everything works and no anxiety, and I want to get mu app V1 out based on Vue 2, then once Vue 3 is more mature I'll then convert my app. 
Right now everything works...

**Comment by KevinExtremo** [comment]
> @KevinExtremo ;
> 
> I'll share my two cents with you.
> I've been going through this whole thing about should I start with Vue 2 or 3 so many times, and after literally creating a comparison list and reflecting on years of being on cutting/bleeding edge, this is the first time I feel I'll wait for Vue 3.1 to come out to make sure all supporting libraries have caught up to Vue 3. It's not just Quasar, but a Vue project uses so many internal libraries.
> 
> Right now I'm building an app with Vue 2 and Quasar 1.x and everything works and no anxiety, and I want to get mu app V1 out based on Vue 2, then once Vue 3 is more mature I'll then convert my app.
> Right now everything works...

You know what Ben, you make a very valid point. Thank you so much for your input. There are always libraries that matter and they tend not to support Vue3 either which can be a true pain.

I've admittedly not really given that any thought, since the app I am intending to make is very simple and for personal use.. but maybe that simple fact alone should speak for me not being too concerned about what Vue version I use?

Thanks once again for that feedback. I think my curiosity and excitement to try Vue3 just got the better of me :-D

**Comment by bzd2000** [comment]
> > @KevinExtremo ;
> > I'll share my two cents with you.
> > I've been going through this whole thing about should I start with Vue 2 or 3 so many times, and after literally creating a comparison list and reflecting on years of being on cutting/bleeding edge, this is the first time I feel I'll wait for Vue 3.1 to come out to make sure all supporting libraries have caught up to Vue 3. It's not just Quasar, but a Vue project uses so many internal libraries.
> > Right now I'm building an app with Vue 2 and Quasar 1.x and everything works and no anxiety, and I want to get mu app V1 out based on Vue 2, then once Vue 3 is more mature I'll then convert my app.
> > Right now everything works...
> 
> You know what Ben, you make a very valid point. Thank you so much for your input. There are always libraries that matter and they tend not to support Vue3 either which can be a true pain.
> 
> I've admittedly not really given that any thought, since the app I am intending to make is very simple and for personal use.. but maybe that simple fact alone should speak for me not being too concerned about what Vue version I use?
> 
> Thanks once again for that feedback. I think my curiosity and excitement to try Vue3 just got the better of me :-D

I also couldn't make up my mind to start my app on vue 3 directly instead of vue 2 some months ago.  
Spend a lot of time looking for libraries already supporting vue 3.  
To finally come back to quasar V1 because nothing is yet stable enough.
Too many libraries I wanted to use, just don't support vue 3 yet. 
So finally it is better to start it in vue 2 and upgrade it later when all libraries (like quasar) are ready.
I do use already the composition api to hopefully make the migration easier.

**Comment by hawkeye64** [comment]
The whole Vue eco-system is going to take a year or more to catch up to Vue v3

**Comment by spinlokd** [comment]
> > > @KevinExtremo ;
> > > I'll share my two cents with you.
> > > I've been going through this whole thing about should I start with Vue 2 or 3 so many times, and after literally creating a comparison list and reflecting on years of being on cutting/bleeding edge, this is the first time I feel I'll wait for Vue 3.1 to come out to make sure all supporting libraries have caught up to Vue 3. It's not just Quasar, but a Vue project uses so many internal libraries.
> > > Right now I'm building an app with Vue 2 and Quasar 1.x and everything works and no anxiety, and I want to get mu app V1 out based on Vue 2, then once Vue 3 is more mature I'll then convert my app.
> > > Right now everything works...
> > 
> > 
> > You know what Ben, you make a very valid point. Thank you so much for your input. There are always libraries that matter and they tend not to support Vue3 either which can be a true pain.
> > I've admittedly not really given that any thought, since the app I am intending to make is very simple and for personal use.. but maybe that simple fact alone should speak for me not being too concerned about what Vue version I use?
> > Thanks once again for that feedback. I think my curiosity and excitement to try Vue3 just got the better of me :-D
> 
> I also couldn't make up my mind to start my app on vue 3 directly instead of vue 2 some months ago.
> Spend a lot of time looking for libraries already supporting vue 3.
> To finally come back to quasar V1 because nothing is yet stable enough.
> Too many libraries I wanted to use, just don't support vue 3 yet.
> So finally it is better to start it in vue 2 and upgrade it later when all libraries (like quasar) are ready.
> I do use already the composition api to hopefully make the migration easier.

Sorry for a side-track question, but I see a number of people comment about how it will take a long time before Vue-niverse catches up to changes in Vue 3 and that most libraries are not ready yet. What are some of the must-have Vue libraries you guys rely on?
Coming from the Java world, I typically try to minimize my dependency stack as I have spent too much time cleaning the maven dependency list or it becomes a nightmare for long-term maintenance/support. Mind you, I do realize that I am using practices from a totally different development realm When I started working on the proof of concept of my vue app (still in vue2), I focused on Quasar, axios and lodash, figuring these would be enough. Granted, the poc has no form validation or proper security, but I imagine these are utility libraries, that probably wouldn't get impacted by vue2/vue3 differences.
I also tried to stay away from any additional visual element libraries, fearing that they would result in a non-homogeneous UI.
Now I am waiting for Quasar beta1 to try to evolve my vue2-based poc to vue 3.
So if you think there are some use-or-die libraries out there that are worth waiting for over getting an early start with v3, please let me know...

**Comment by hawkeye64** [comment]
@spinlokd It's going to take a a while before any Quasar app-extensions catch up. Even if Quasar is Vue 3 compliant, it will take more time for app-extensions you might rely on. Even aside from Quasar, I still see Vue 2 plugins not available, when you think they should be. The point is, no one should think that everything is going to catch-up overnight. If you believe that, then you shoiuld be sponsoring them or you really have nothing to say that has an impact.

**Comment by DaDuFan** [comment]
These are some good arguments to prefer V1, for the moment.
But, what will happen when V2 is released ?
How long will V1 be supported. Bug-fixes ? Improvements ?

I read, that there will be a migration guide. But when Quasar app-extensions are not compliant, yet, this can be a migration stopper.

Will the Quasar app-extensions interface change, or will it stay the same ?

Thanks, for this great framework. Definitely worth sponsoring .

**Comment by IlCallo** [comment]
Hey there!

> How long will V1 be supported. Bug-fixes ? Improvements ?

For sure until the end of 2021, probably more, still have to discuss this properly.
When time permits, we'll also explore if it'll be feasible to create a compibility layer to make Quasar 2 support Vue2 codebases.
There are some packages of Vue ecosystem (eg. `vue-demi`, etc) under active development right now which could ease or prevent the need of a manual transition, we're waiting for those to get a bit more mature.
Vue core team is also developing an automatic migration tool.

> Will the Quasar app-extensions interface change, or will it stay the same ?

The interface will be the same, but AE code must be adapted to support Vue3 (eg. update deps) most of the time.
Some AE are already being worked on, we'll share details with you as soon as we have news on this topic

**Comment by q389491729** [comment]
Maybe  can use the  `` quasar create <folder_name> --branch next `` command  to try the alpha version and experience vue3

**Comment by rstoenescu** [comment]
Beta 1 release is imminent. Today (Feb 2nd) or tomorrow the latest.

**Comment by xderorschach** [comment]
🎉👍Waiting for SSR!

**Comment by rstoenescu** [comment]
Took us a little more than expected, but Quasar v2 beta with Vue 3 support is finally out 🥳 🥳 🥳 🥳 🥳  🎉 🎉 🎉 🎉 🎉 🎉 

We're eager to know what do you think, jump in! 
https://next.quasar.dev

**Comment by JulesNeck** [comment]
The champagne is already chilled. I'm going to party now! 🍾🍸🥂 🎂  
Huge thanks to Razvan and the whole team! 🏆🏅

**Comment by stianro** [comment]
Fantastic. You guys are the best. Great progress on a awesome product which fell in love with back in the 0.x days 🙂

**Comment by BenHayat** [comment]
So guys, as some of us were discussing about staying with Vue 2 for some time, I have a question.
Personally, I've found Vuex to be too verbose and adds too much code to a large app and I find Composition to be more natural with my thinking as it's very close to .Net state management system and requires a lot less code. Honestly, the ONLY reason I was considering Vue 3 was to use composition API. And now we have a plugin to use Composition Api in Vue 2.

My question is, is it safe to go this route using this plugin with Vue 2 and use Vuex in lesser capacity?

Thanks!

**Comment by IlCallo** [comment]
@BenHayat not really a thing related to Quasar 2 release, but here you can find some alternatives to Vuex:
- https://gist.github.com/IlCallo/e01468917d7b79acf5f5fc1ab4f253b9
- https://dev.to/quasar/state-management-without-vuex-or-other-depedencies-28np

**Comment by mycarrysun** [comment]
> My question is, is it safe to go this route using this plugin with Vue 2 and use Vuex in lesser capacity?

You don't have to use Vuex for everything and honestly you shouldn't...only if you need to access the data in multiple separated areas of your app would you need to use something like Vuex. I personally have been using it less and less because it makes my app feel faster.

**Comment by BenHayat** [comment]
> 
> 
> > My question is, is it safe to go this route using this plugin with Vue 2 and use Vuex in lesser capacity?
> 
> You don't have to use Vuex for everything and honestly you shouldn't...only if you need to access the data in multiple separated areas of your app would you need to use something like Vuex. **I personally have been using it less and less because it makes my app feel faster.**

I agree with you, as Vuex creates too many layers of code execution which affects the performance.
I suppose in cases like where a component needs use parent data is to pass data to component than using Vuex.

I haven't a clear cut guidelines when to use and when NOT to use Vuex.

**Comment by mycarrysun** [comment]
> I haven't seen a clear cut guidelines when to use and when NOT to use Vuex.

I'm sure someone has probably written a blog article or two on this but in my mind if I need access to something that is somewhat global (like user information, config data, etc) I like to put it in the Vuex store. Otherwise if you can get away with loading things in from a network request on mount of a component (and you're not making the request very often) that is the way to go. If the component is making a lot of the same network request on mount, it might be better to just store in the Vuex store and only update it from the network when necessary. Those are the main use-cases I can think of when it's better to put in Vuex.

**Comment by BenHayat** [comment]
> 
> 
> > I haven't seen a clear cut guidelines when to use and when NOT to use Vuex.
> 
> I'm sure someone has probably written a blog article or two on this but in my mind **if I need access to something that is somewhat global (like user information, config data, etc) I like to put it in the Vuex store**. **_Otherwise if you can get away with loading things in from a network request on mount of a component (and you're not making the request very often) that is the way to go._** 

Thank you for your comment.
For my user data and system config data, I use the Vuex and it's fine.
But I need to evaluate if multiple components making their own network calls to the same REST endpoint, I should then use Vuex to avoid multiple calls.

Note: I had found a few blogs on the use of Vuex, but they were too simplistic case as for a demo app than a "Real" world situation.

I thank you again for your responses.

**Comment by mycarrysun** [comment]
> But I need to evaluate if multiple components making their own network calls to the same REST endpoint, I should then use Vuex to avoid multiple calls.

Yea 1 or 2 probably doesn't deserve Vuex, but 3 is when I would start thinking about putting in Vuex instead of getting it inside the component. It does add additional complexity if you want to reduce the network requests because you have to then check on the frontend if you need to get a new response from the server but at least you will have a net improvement in the amount of network requests.

**Comment by Milancos** [comment]
I think someone should lock this theme because people are going WAAY OFFTOPIC

**Comment by hawkeye64** [comment]
Yes, please. Everyone take your discussions to either Discord channels or Forums. Keep this discussion related to the Beta Release **only**. Thank you.

**Comment by Focushift** [comment]
> We're eager to know what do you think, jump in!
> https://next.quasar.dev

Not working in FF. Regex error in console.

**Comment by pdanpdan** [comment]
@Focushift Thank you, very useful comment :D
It is working just fine in Firefox

**Comment by danielfaust** [comment]
@Focushift works fine for me as well in FF. With uBlock Origin and stuff. No problems.

**Comment by JulesNeck** [comment]
Works fine in FF on my machine as well.

**Comment by Focushift** [comment]
> 
> 
> @Focushift Thank you, very useful comment :D
> It is working just fine in Firefox

Okay

v75 64 bit with uBlock and etc
Win 10 1909
Still not working

SyntaxError: invalid regular expression flag s
app.baba1da8.js:1:147079

`return a+t.replace(/>/g,"___TEMP_REPLACEMENT___")+o})).replace(/<(q-[\w-]+|div)([^>]*?)\s*?([\n\r][\t ]+)?\/>/gs,"<$1$2$3></$1>")`

seems like he dont like `\s`

**Comment by pdanpdan** [comment]
Yes, that is known, but it's only related to the docs site. Try to use a browser that has better support for regex when you read the docs (Firefox 78+)

**Comment by rstoenescu** [comment]
Quick update: SSR is ready (UI and CLI-wise) to be released. Unfortunately, there are quite a few Vue SSR bugs that need to be fixed before we enable the SSR mode for Quasar v2. We are in the process of reporting them and we'll start collaborating with the Vue team to help fix them.

**Comment by hawkeye64** [comment]
Amazing work @rstoenescu like a painter on a canvas :)

**Comment by smolinari** [comment]
More like Michelangelo on the Sistine Chapel. 😁 

Scott

**Comment by rstoenescu** [comment]
Another quick update.

We're close to the stable release. The only thing remaining is the SSR support. Quasar is SSR ready, however we're only going enable SSR support as soon as these Vue3 SSR bugs (critical for us and your experience) are solved:
* https://github.com/vuejs/vue-next/issues/3244
* https://github.com/vuejs/vue-next/issues/2386
* https://github.com/vuejs/vue-next/issues/3245
* https://github.com/vuejs/vue-next/issues/3243

**Comment by rstoenescu** [comment]
Reported one more vue3 ssr bug to Vue Team: https://github.com/vuejs/vue-next/issues/3255

**Comment by rstoenescu** [comment]
## Is v2 stable?
We are currently waiting on Vue Team to fix a few tickets for Vue core that we raised (see the February 16th update) in order to enable SSR mode and release the stable version. Otherwise, you can consider the latest beta as "stable".

**Comment by bambuschop** [comment]
What about issue https://github.com/quasarframework/quasar/issues/8534 - without HMR v2 for me and my company can't be considered stable since manually reloading all the time gets tiring. 😅

**Comment by rstoenescu** [comment]
@bambuschop That is an issue that should be reported to `vue-class-component` owners, not Quasar.

**Comment by bichikim** [comment]
all!!!!! SSR issues are closed @rstoenescu  🎉🎉🎉🎉😂

**Comment by rstoenescu** [comment]
Indeed. There's still a Vue issue with keep-alive and child component using the default slot (will log a ticket for Vue on this as well). But this one has limited impact as opposed to all other Vue tickets, so we can now at least enable SSR in the App CLI. Expect the magic to happen near the end of this week. Polishing everything SSR-wise at the moment.

**Comment by lukaszraczylo** [comment]
@rstoenescu Sweet! Thank you so much!

**Comment by Liwoj** [comment]
Just  a warning for all early adopters...
I'v upgraded to Quasar 2.0.0-beta.11 few days ago + Vue v3.0.7 
After upgrading to Vue v3.0.9, `QDialog` is broken due to Vue issue [#3497](https://github.com/vuejs/vue-next/issues/3497)

**Comment by rstoenescu** [comment]
> Just  a warning for all early adopters...
> 
> I'v upgraded to Quasar 2.0.0-beta.11 few days ago + Vue v3.0.7 
> 
> After upgrading to Vue v3.0.9, `QDialog` is broken due to Vue issue [#3497](https://github.com/vuejs/vue-next/issues/3497)

This is why you should always use the vue version shipped by Quasar App CLI and not install it manually.

**Comment by weiweilipl** [comment]
We need SSR. When will the stable version v2 be released

**Comment by yanghuilong** [comment]
ssr issues are closed! quasar2 stable ?

**Comment by IlCallo** [comment]
@yanghuilong we'll release SSR support, then we'll need some time for that part to be tested by the community before releasing Quasar2 as stable :)

Also look previous posts, quoting Razvan:
> Is v2 stable?
> We are currently waiting on Vue Team to fix a few tickets for Vue core that we raised (see the February 16th update) in order to enable SSR mode and release the stable version. Otherwise, you can consider the latest beta as "stable".

**Comment by rstoenescu** [comment]
The last Quasar CLI mode left to be enabled, which is SSR, is now also available.
Here is the link to the [upgrade guide](https://next.quasar.dev/quasar-cli/developing-ssr/ssr-upgrade-guide) on SSR.
We are targeting for a 2 weeks testing period for SSR mode then we will release the official stable version of Quasar v2.

**Comment by lukaszraczylo** [comment]
@rstoenescu there goes my night... :) Thank you a million!

**Comment by rstoenescu** [comment]
@lukaszraczylo Many sleepless nights went into this release. So yeah, this is my "revenge" :P :P :P

**Comment by toralux** [comment]
Anyway, using @quasar/cli@1.1.3 preparing for SSR (https://next.quasar.dev/quasar-cli/developing-ssr/preparation) i get this message: "SSR mode is not *yet* supported and is disabled currently." Is cli support coming or should I prepare for SSR manually for now?

**Comment by IlCallo** [comment]
@toralux have you upgraded `@quasar/app` to the latest beta version?

**Comment by toralux** [comment]
@IlCallo Thanks! Sorry, my bad ... "Creating SSR source folder, SSR support was added"!

**Comment by 3kynox** [comment]
Hi! Lot of works ongoing on webpack5. Very nice @rstoenescu !

**Comment by rstoenescu** [comment]
@3kynox not only that. lots of perf improvements on q/app too!

**Comment by soulsam480** [comment]
Hii, any update on Quasar testing for V2 ?

**Comment by rstoenescu** [comment]
@soulsam480 it's still not ready yet, unfortunately.

**Comment by rstoenescu** [comment]
Small but important update:

We have passed the final major milestone: Quasar CLI now ships with Webpack 5, which was badly needed to support Node 13+. Faster build speeds and lower memory usage.

We're just a few small steps away from getting Quasar v2 out of beta!

**Comment by francoism90** [comment]
@rstoenescu This is what happens when using V2 on a new (clean) project, seems something is broken:
```
$ quasar create test --branch next
$ cd test
$ quasar build

 Build mode........ spa
 Pkg quasar........ v2.0.0-beta.13
 Pkg @quasar/app... v3.0.0-beta.17
 Pkg webpack....... v5
 Debugging......... no
 Publishing........ no

 Configured browser support (at least 88.20% of global marketshare):
 · Chrome for Android >= 90
 · Firefox for Android >= 87
 · Android >= 90
 · Chrome >= 80
 · Edge >= 87
 · Firefox >= 79
 · iOS >= 11.0-11.2
 · Opera >= 71
 · Safari >= 11

 App • Chaining "UI" Webpack config
 App • Cleaned build artifact: "/path/test/dist/spa"
 App •  WAIT  • Compiling of "UI" in progress...

 • Compiling:
 └── UI ██                   7% setup compile ExternalsPlugin

/path/test/node_modules/webpack/lib/NormalModule.js:181
			throw new TypeError(
			      ^

[  TypeError: The 'compilation' argument must be an instance of Compilation
  
  - NormalModule.js:181 Function.getCompilationHooks
    [test]/[webpack]/lib/NormalModule.js:181:10
  
  - pluginWebpack5.js:37 
    [test]/[vue-loader]/dist/pluginWebpack5.js:37:26
  
  
  - Hook.js:14 Hook.CALL_DELEGATE [as _call]
    [test]/[tapable]/lib/Hook.js:14:14
  
  - Compiler.js:1017 Compiler.newCompilation
    [test]/[app]/[webpack]/lib/Compiler.js:1017:26
  
  - Compiler.js:1059 
    [test]/[app]/[webpack]/lib/Compiler.js:1059:29
  
  
  - Hook.js:18 Hook.CALL_ASYNC_DELEGATE [as _callAsync]
    [test]/[tapable]/lib/Hook.js:18:14
  
  - Compiler.js:1054 Compiler.compile
    [test]/[app]/[webpack]/lib/Compiler.js:1054:28
  
  - Compiler.js:494 
    [test]/[app]/[webpack]/lib/Compiler.js:494:12
  
]
```

**Comment by hawkeye64** [comment]
@francoism90 Just tested and had no issues

```bash
$ quasar create q2-with-wp5 --branch next

  ___                             
 / _ \ _   _  __ _ ___  __ _ _ __ 
| | | | | | |/ _` / __|/ _` | '__|
| |_| | |_| | (_| \__ \ (_| | |   
 \__\_\\__,_|\__,_|___/\__,_|_|   
 


? Project name (internal usage for dev) q2-with-wp5
? Project product name (must start with letter if building mobile apps) Quasar A
pp
? Project description A Quasar Framework app
? Author Jeff Galbraith <galbraith64@gmail.com>
? Pick your CSS preprocessor: SCSS
? Check the features needed for your project: ESLint (recommended)
? Pick an ESLint preset: Prettier
? Continue to install project dependencies after the project has been created? (
recommended) yarn

  Quasar CLI · Generated "q2-with-wp5".



 [*] Installing project dependencies ...

yarn install v1.22.5
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning " > @babel/eslint-parser@7.13.14" has unmet peer dependency "@babel/core@>=7.11.0".
warning " > eslint-webpack-plugin@2.5.4" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
[5/5] Building fresh packages...
success Saved lockfile.
Done in 29.02s.


 [*] Running eslint --fix to comply with chosen preset rules...


yarn run v1.22.5
$ eslint --ext .js,.vue ./ --fix
Done in 1.49s.

 [*] Quasar Project initialization finished!

To get started:

  cd q2-with-wp5
  quasar dev

Documentation can be found at: https://quasar.dev

Quasar is relying on donations to evolve. We'd be very grateful if you can
read our manifest on "Why donations are important": https://quasar.dev/why-donate
Donation campaign: https://donate.quasar.dev
Any amount is very welcomed.
If invoices are required, please first contact razvan@quasar.dev

Please give us a star on Github if you appreciate our work:
https://github.com/quasarframework/quasar

Enjoy! - Quasar Team


jeff@jeff-UX510UX:/data2/projects/sandbox$ 
jeff@jeff-UX510UX:/data2/projects/sandbox$ cd q2-with-wp5/
jeff@jeff-UX510UX:/data2/projects/sandbox/q2-with-wp5$ quasar build

 Build mode........ spa
 Pkg quasar........ v2.0.0-beta.13
 Pkg @quasar/app... v3.0.0-beta.17
 Pkg webpack....... v5
 Debugging......... no
 Publishing........ no

 Configured browser support (at least 88.20% of global marketshare):
 · Chrome for Android >= 90
 · Firefox for Android >= 87
 · Android >= 90
 · Chrome >= 80
 · Edge >= 87
 · Firefox >= 79
 · iOS >= 11.0-11.2
 · Opera >= 71
 · Safari >= 11

 App • Chaining "UI" Webpack config
 App • Cleaned build artifact: "/data2/projects/sandbox/q2-with-wp5/dist/spa"
 App •  WAIT  • Compiling of "UI" in progress...
 App •  DONE  • "UI" compiled with success • 11476ms
 App • Added build artifact "/data2/projects/sandbox/q2-with-wp5/dist/spa"

 Summary for "UI" (only css/js/json)
 ╔═════════════════════╤═══════════╤══════════╗
 ║               Asset │      Size │  Gzipped ║
 ╟─────────────────────┼───────────┼──────────╢
 ║    app.31d6cfe0.css │   0.00 KB │  0.02 KB ║
 ║ vendor.7f4564c6.css │ 185.82 KB │ 33.20 KB ║
 ╟─────────────────────┼───────────┼──────────╢
 ║     193.e9f54f5c.js │   0.72 KB │  0.47 KB ║
 ║     717.56310ac6.js │   3.45 KB │  1.38 KB ║
 ║     744.16eb4d30.js │   0.55 KB │  0.39 KB ║
 ║     app.09b8285d.js │   3.61 KB │  1.83 KB ║
 ║  vendor.4a58b838.js │ 162.30 KB │ 57.65 KB ║
 ╚═════════════════════╧═══════════╧══════════╝
 
 Build succeeded

 Build mode........ spa
 Pkg quasar........ v2.0.0-beta.13
 Pkg @quasar/app... v3.0.0-beta.17
 Pkg webpack....... v5
 Debugging......... no
 Publishing........ no
 Transpiled JS..... yes (Babel)
 ==================
 Output folder..... /data2/projects/sandbox/q2-with-wp5/dist/spa

 Tip: Built files are meant to be served over an HTTP server
      Opening index.html over file:// won't work

 Tip: You can use "$ quasar serve" command to create a web server,
      both for testing or production. Type "$ quasar serve -h" for
      parameters. Also, an npm script (usually named "start") can
      be added for deployment environments.
      If you're using Vue Router "history" mode, don't forget to
      specify the "--history" parameter: "$ quasar serve --history"
```

**Comment by francoism90** [comment]
@hawkeye64 Thanks, I'll try tomorrow once again. :)

**Comment by Smrtnyk** [comment]
works fine for me too

**Comment by francoism90** [comment]
Turns out this is happening because `standardjs`, this also seems to broken the process of installing packages.
I'll switch to Prettier as this seems to be used by most developers a well. :)

**Comment by francoism90** [comment]
After switching to Prettier things work OK. However building/running a capacitor app doesn't:
```
native App.addListener (#59478500)
vendor.js:3 ReferenceError: onMounted is not defined
    at setup (app.js:1)
    at o (vendor.js:3)
    at En (vendor.js:3)
    at Tn (vendor.js:3)
    at D (vendor.js:3)
    at N (vendor.js:3)
    at y (vendor.js:3)
    at Q (vendor.js:3)
    at mount (vendor.js:3)
    at Object.t.mount (vendor.js:3)
```
It seems to be caused by the dialog plugin, can anyone confirm this?

**Comment by rstoenescu** [comment]
@francoism90 fixed and released q/app beta.18

**Comment by luozhihua** [comment]
SSR is posible to support KoaJS ?

**Comment by ekoeryanto** [comment]
@rstoenescu beta 18 not fixing setup with airbnb

**Comment by dioslibre** [comment]
@francoism90 How did you resolve the compilation issue? what do you mean by " switching to Prettier"?

**Comment by francoism90** [comment]
@dioslibre Don't know if this is the answer you're looking for. Anyway, this is how I generally upgrade between Quasar V2 beta's:
- Current project is on older beta (e.g.  q/app beta.12)
- Create a new project using `quasar create temp_project --branch next` and select the same options as the ones I'm currently using (or in this case changed the ES preset to Prettier instead of Standard).
- Compare the current and newly created project (Meld is a good tool), merge/change/fix to current project when needed.

Switching to Prettier required me to rewrite files like `babel.config.js`, `.eslintrc.js`, etc. I have done the same thing on every (big) Quasar upgrade or when issues are popping up. Quasar does a great job in keeping 'system' files out of your project, however sometimes you may need to sync them.

**Comment by 3kynox** [comment]
Using the normal way to upgrade Quasar could be more convenient:
`quasar upgrade --install`

from docs: https://next.quasar.dev/start/upgrade-guide#older-v2-to-latest-v2

**Comment by dioslibre** [comment]
@francoism90 The problem happens even on a newly created project, weird.

**Comment by francoism90** [comment]
@3kynox True, I use `quasar upgrade -i` to perform upgrades, `ncu` to check for other dep updates, and if needed use meld to compare stuff. In this case it was needed because of the eslint preset change.

**Comment by memphisdj** [comment]
One interesting thing is, if i create a new project and install using yarn it works, but if i install it using npm, once i try to build, it fails with **The 'compilation' argument must be an instance of Compilation**.

**Comment by francoism90** [comment]
@memphisdj It works fine with npm here, what's your npm version?

**Comment by memphisdj** [comment]
> @memphisdj It works fine with npm here, what's your npm version?

NPM 7.9.0

**Comment by smolinari** [comment]
We always highly recommend using Yarn for local dependency management. NPM for global packages only. 

Scott

**Comment by memphisdj** [comment]
Hi Scott, I do always use yarn, but I just decided to build with npm and the error happened, so I'm reporting it. If it's available as a choice in the cli, in my opinion it should work. 

Marin

**Comment by smolinari** [comment]
@memphisdj - Sure. But, please report it in its own issue, so it can get the necessary attention. Thanks.

Scott

**Comment by therealcoder1337** [comment]
https://github.com/quasarframework/quasar/issues/7836#issuecomment-829052427
> One interesting thing is, if i create a new project and install using yarn it works, but if i install it using npm, once i try to build, it fails with **The 'compilation' argument must be an instance of Compilation**.

same issue: https://github.com/quasarframework/quasar/issues/9092

seems like a bug that only occurs with npm 7. it should work with npm 6 or yarn, though.

**Comment by jonathanpmartins** [comment]
> [#7836 (comment)](https://github.com/quasarframework/quasar/issues/7836#issuecomment-829052427)
> 
> seems like a bug that only occurs with npm 7. it should work with npm 6 or yarn, though.

Thanks, moving to yarn!

**Comment by jonathanpmartins** [comment]
> [#7836 (comment)](https://github.com/quasarframework/quasar/issues/7836#issuecomment-829052427)
> 
> > One interesting thing is, if i create a new project and install using yarn it works, but if i install it using npm, once i try to build, it fails with **The 'compilation' argument must be an instance of Compilation**.
> 
> same issue: #9092
> 
> seems like a bug that only occurs with npm 7. it should work with npm 6 or yarn, though.

I don't thing this is a NPM 7 issue. This problem was possible introduced with the ultimate releases of the beta version (remember is in development).

I had a worked app until I update it to the last beta version.

Working with NPM 7:
```json
"dependencies": {
  "quasar": "2.0.0-beta.12",
},
"devDependencies": {
  "@quasar/app": "3.0.0-beta.15",
},
```

Not Working with NPM 7:
```json
"dependencies": {
  "quasar": "2.0.0-beta.15",
},
"devDependencies": {
  "@quasar/app": "3.0.0-beta.20",
},
```

**Comment by jacksonGross** [comment]
I've been following along this thread around the support of Vue 3 in Quasar V2 and am wondering if there are any plans to support the migration/compat build described here? https://github.com/vuejs/vue-next/tree/master/packages/vue-compat

It seems like it could be a good option to help migrate some of the more larger code bases incrementally as there are quite a number of deprecations and [breaking changes](https://v3.vuejs.org/guide/migration/introduction.html#breaking-changes) in Vue 3.

**Comment by rstoenescu** [comment]
Forgot to post some updates here. So:
* the NPM v7 issue is solved
* we'll try to write a guide on vue-compat (@jacksonGross )
* creating a project folder with any of the available options will now not result in a lint error

Preparing v2.0.0-rc.1

**Comment by francoism90** [comment]
@rstoenescu Quasar V2 looks good so far! I'm using the AirBNB linter, capacitor and loving the Vue composite API.

**Comment by jonathanpmartins** [comment]
I'm just passing by to congratulate you for the excellent work!

**Comment by rabbiveesh** [comment]
> Forgot to post some updates here. So:
> 
> * the NPM v7 issue is solved
> * we'll try to write a guide on vue-compat (@jacksonGross )
> * creating a project folder with any of the available options will now not result in a lint error
> 
> Preparing v2.0.0-rc.1

Is there any new on the vue-compat guide? I'm itching to try out v2, and need to do the migration build so I can upgrade smoothly

**Comment by soulsam480** [comment]
Amazing work @rstoenescu

**Comment by francoism90** [comment]
@rabbiveesh How exactly does it work? I rewrote the project in Vue3, just because the Composite API works so good and I could remove ugly things like mixins.

I was very skeptical on the Vue3 changes, but so far I really  like them. :)

**Comment by IlCallo** [comment]
Small heads up on what has been happening lately regarding Quasar v2 release.

We have been postponing Qv2 stable release (eg. we planned to release RC on 28th of May) due to some last minute problems with some upstream deps.

In particular, latest Sass releases added deprecation warnings for some widely used features, and this generates A LOT of warnings.
Even if it's not technically a breaking change, it's an horrible DX not related to Quasar.
To avoid it, we would need to refactor most of Quasar CSS, which will result into a breaking change for Qv2 early adopters which went through the beta, and we don't like this option.

On the other side, there seems to be no way to silence those warnings yet.
There is an undergoing effort to silence at least the warnings coming from Quasar code itself (https://github.com/sass/sass/issues/3065), but this solves only half of the problem.

Anyway, we're now into RC phase, we'll continue to work to solve this and other issues during RC phase, thanks for your patience and support <3

**Comment by xizprodev** [comment]
Is there a problem with quasar vue 3 and webserver? 
On my old admin build on quasar with vue 2 i have this configuration for devserver:
devServer: {
      public: 'https://admin.local',
      https: false,
      port: 8080,
      open: false // opens browser window automatically
    },
And it works very good, but when i upgrade to quasar vue 3 (i create a new project with quasar create --branch next and move some files from old src to new src folder) i got lot of errors in console WebSocket connection to 'wss://admin.local:8080/ws' failed:

**Comment by IlCallo** [comment]
Hi @xizprodev may I ask you to create a repro and open a new issue?
Underlying webpack-dev-server advanced by a version and some things changed, so you should probably start by checking that out if you have problems with it

**Comment by maggie44** [comment]
> Is there a problem with quasar vue 3 and webserver?
> On my old admin build on quasar with vue 2 i have this configuration for devserver:
> devServer: {
> public: 'https://admin.local',
> https: false,
> port: 8080,
> open: false // opens browser window automatically
> },
> And it works very good, but when i upgrade to quasar vue 3 (i create a new project with quasar create --branch next and move some files from old src to new src folder) i got lot of errors in console WebSocket connection to 'wss://admin.local:8080/ws' failed:

Check out https://github.com/quasarframework/quasar/issues/9466

**Comment by rulrok** [comment]
Just a curiosity: is the performance improvement of `q-img` from v2 going to be ported eventually to v1?
From what I read it seemed quite important to just be left out.

**Comment by IlCallo** [comment]
QImg optmizations (as others Quasar v2 optimizations) rely on modern features which IE doesn't support.
Since Qv1 must be IE compatible, it's not possible to backport those optmizations

**Comment by rstoenescu** [comment]
Preparing for the v2.0.0 stable release.
Branch `vue3-work` has been merged into `dev`. The Qv1 work will now be done exclusively on the new `v1` branch.

**Comment by rstoenescu** [comment]
The [big announcement](https://github.com/quasarframework/quasar/discussions/9734) has come. Time to close this ticket 🥇

**Comment by rodrigoslayertech** [comment]
My old project is using multiple Quasar extensions. I can't migrate yet. Quasar Extensions are taking a long time to port Quasar 2.0.

**Comment by francoism90** [comment]
@slowaways What kind of entensions are you using? I have migrated to other solutions when the extension wasn't available anymore.

**Comment by rodrigoslayertech** [comment]
@francoism90 QIconPicker, QPDFViewer, QCalendar...

**Comment by francoism90** [comment]
@slowaways 

https://github.com/quasarframework/app-extension-qpdfviewer/releases/tag/v2.0.0-alpha.2
https://github.com/quasarframework/quasar-ui-qcalendar/releases/tag/v4.0.0-beta.4

QIconPicker seems to be compatible and maintained?

**Comment by rodrigoslayertech** [comment]
@francoism90 Yes, but waiting release version.

**Comment by francoism90** [comment]
@slowaways You could already start developing in Quasar v2 as most extension breaking changes will probably be minor. I also started with the beta's (including non Quasar extensions) and could easily move to the stable builds when they become available.

**Comment by hawkeye64** [comment]
@slowaways QIconPicker (@next branch) will be released soon

**Comment by melroy89** [comment]
Thank you so much!

**Comment by zenflow** [comment]
https://github.com/quasarframework/quasar/issues/7836#issuecomment-855291820

> we'll try to write a guide on vue-compat (@jacksonGross )

@rabbiveesh @jacksonGross Any guide/resources on this? Or any luck implementing it? (i.e. is it possible?)

**Comment by IlCallo** [comment]
@zenflow 
Have you checked out https://github.com/vuejs/core/tree/main/packages/vue-compat#readme ?
If yes, did you had any problems with it? It's not clear from your question

**Comment by zenflow** [comment]
@IlCallo Yup, I have. I was just wondering if there were any additional tips or info (caveats?) on doing it with Quasar. I'm attempting it now, so I will report back here, and open issues if I come across any.

**Comment by IlCallo** [comment]
Thanks! We appreciate when people give back to the community :D
If there's a dedicated issue, please use that one to post feedback.
If there isn't and you hit some problems, please create a new one so we can avoid spamming all the people subscribed to this issue
